### PR TITLE
Move to Shared-MetadataUtilsSmi

### PR DESCRIPTION
**Pull request by Kaur-Parminder**
This is part of #1261 Move the file file common folder.

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Server/MetadataUtilsSmi.cs [hunk 1]
```diff
@@ -224,6 +232,10 @@ internal static bool IsPlpFormat(SmiMetaData metaData)
                             extendedCode = ExtendedClrTypeCode.Boolean;
                         else if (value.GetType() == typeof(SqlBoolean))
                             extendedCode = ExtendedClrTypeCode.SqlBoolean;
+#if NETFRAMEWORK
+                        else if (Type.GetTypeCode(value.GetType()) == TypeCode.Boolean)
+                            extendedCode = ExtendedClrTypeCode.Boolean;
+#endif
```
Comment (DavoudEshtehari) [inline src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Server/MetadataUtilsSmi.cs:238]: For instance, this seems redundant checking.

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Server/MetadataUtilsSmi.cs [hunk 2]
```diff
@@ -439,7 +500,8 @@ internal static SqlMetaData SmiExtendedMetaDataToSqlMetaData(SmiExtendedMetaData
                     source.TypeSpecificNamePart2,
                     source.TypeSpecificNamePart3,
                     true,
-                    source.Type);
+                    source.Type
```
Comment (DavoudEshtehari) [inline src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Server/MetadataUtilsSmi.cs:503]: Could you check if "SmiExtendedMetaDataToSqlMetaData" function hasn't been used and could be removed?

**Review by DavoudEshtehari — COMMENTED** [review]
Could you include the inline comments in XML summary tags where ever it's applicable, please?

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Server/MetadataUtilsSmi.cs (hunk 2 above)
Comment (Kaur-Parminder) [reply src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Server/MetadataUtilsSmi.cs]: It was not being used. I removed it.

File: src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj [hunk 3]
```diff
@@ -338,6 +338,7 @@
     </Compile>
     <Compile Include="..\..\src\Microsoft\Data\SqlClient\SqlCommandBuilder.cs">
       <Link>Microsoft\Data\SqlClient\SqlCommandBuilder.cs</Link>
+      <SubType>Component</SubType>
```
Comment (lcheunglci) [inline src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj:341]: You want to remove this even though visual studios keeps adding it back in.

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj [hunk 4]
```diff
@@ -379,6 +379,12 @@
     <Compile Include="..\..\src\Microsoft\Data\SqlClient\Server\SmiTypedGetterSetter.cs">
       <Link>Microsoft\Data\SqlClient\Server\SmiTypedGetterSetter.cs</Link>
     </Compile>
+    <Compile Include="..\..\src\Microsoft\Data\SqlClient\Server\SmiEventSink.cs">
+      <Link>Microsoft\Data\SqlClient\Server\SmiEventSink.cs</Link>
+    </Compile>
+     <Compile Include="..\..\src\Microsoft\Data\SqlClient\Server\SmiEventSink_Default.cs">
+      <Link>Microsoft\Data\SqlClient\Server\SmiEventSink_Default.Common.cs</Link>
+    </Compile>
```
Comment (lcheunglci) [inline src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj]: I believe we have been placing this alphabetically, so I think this should go above Line 376

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Server/MetadataUtilsSmi.cs [hunk 5]
```diff
@@ -509,8 +499,12 @@ internal static SmiExtendedMetaData SqlMetaDataToSmiExtendedMetaData(SqlMetaData
...
                                             source.Scale,
                                             source.LocaleId,
-                                            source.CompareOptions,
-                                            null,
+                                            source.CompareOptions
+#if NETFRAMEWORK
+                                            ,source.Type,
+#else
+                                            ,null,
+#endif
```
Comment (DavoudEshtehari) [inline src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Server/MetadataUtilsSmi.cs:507]: NIT
```suggestion
                                            source.CompareOptions,
#if NETFRAMEWORK
                                            source.Type,
#else
                                            null,
#endif
```

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Server/MetadataUtilsSmi.cs [hunk 6]
```diff
@@ -526,6 +520,9 @@ internal static bool IsCompatible(SmiMetaData firstMd, SqlMetaData secondMd)
                     firstMd.Scale == secondMd.Scale &&
                     firstMd.CompareOptions == secondMd.CompareOptions &&
                     firstMd.LocaleId == secondMd.LocaleId &&
+#if NETFRAMEWORK
+                    firstMd.Type == secondMd.Type &&
+#endif
```
Comment (DavoudEshtehari) [inline src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Server/MetadataUtilsSmi.cs:525]: Maybe it's been missed in the transition process! Could you verify if we could keep it the same in the both projects , please?
```suggestion
                    firstMd.Type == secondMd.Type &&
```

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj [hunk 7]
```diff
@@ -367,6 +370,12 @@
     <Compile Include="..\..\src\Microsoft\Data\SqlClient\SignatureVerificationCache.cs">
       <Link>Microsoft\Data\SqlClient\SignatureVerificationCache.cs</Link>
     </Compile>
+    <Compile Include="..\..\src\Microsoft\Data\SqlClient\Server\SmiEventSink.cs">
+      <Link>Microsoft\Data\SqlClient\Server\SmiEventSink.cs</Link>
+    </Compile>
+    <Compile Include="..\..\src\Microsoft\Data\SqlClient\Server\SmiEventSink_Default.cs">
+      <Link>Microsoft\Data\SqlClient\Server\SmiEventSink_Default.Common.cs</Link>
+    </Compile>
```
[the remark this answers is not in the document]
Comment (Kaur-Parminder) [reply src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj]: I have reverted it back. Agree #1364 will take care of it.

**Review by DavoudEshtehari — APPROVED** [review]

**Comment by johnnypham** [comment]
There are some lines in netfx (eg [this one](https://github.com/dotnet/SqlClient/blob/c9d59d8426b987e10af245df11853d10bb6a1d0d/src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/Server/MetadataUtilsSmi.cs#L216)) that are missing in the new file. Was there a reason why?

**Review by johnnypham — APPROVED** [review]

**Review by JRahnama — APPROVED** [review]

**Comment by JRahnama** [comment]
Can you address the conflicts please that we can merge the code

**Comment by Kaur-Parminder** [comment]
> Can you address the conflicts please that we can merge the code

Done

**Pull request merged**